src/esperanza/walletextension.h

```diff
@@ -202,6 +202,45 @@ class WalletExtension : public staking::StakingWallet {
       const finalization::VoteRecord &vote2);
 
   void PostInitProcess(CScheduler &scheduler);
+
+  template <typename Callable>
+  void ForEachMatureOutput(
+      const CTransaction &tx,
+      const CBlockIndex *block,
+      blockchain::Depth depth,
+      Callable f) const {
+    std::size_t reward_offset = 0;
+    if (tx.IsCoinBase() && !block) {
```

Comment:
maybe wrap both of these `if` clauses in one bigger `if (tx.IsCoinBase())`?

Reply:
Done.